pkg/tfvar/tfvar.go

```diff
@@ -110,7 +112,62 @@ func WriteAsTFVars(w io.Writer, vars []Variable) error {
 	_, err := f.WriteTo(w)
 	return errors.Wrap(err, "tfvar: failed to write as tfvars")
 }
+func WriteAsWorkspacePayload(w io.Writer, vars []Variable) error {
+	var payload error
+	// type SimpleJSONValue struct {
+	// 	cty.Value
+	// }
 
+	for _, v := range vars {
+		val := convertNull(v.Value)
+
+		t := hclwrite.TokensForValue(val)
+		t = oneliner(t)
+		b := hclwrite.Format(t.Bytes())
+		b = bytes.TrimPrefix(b, []byte(`"`))
+		b = bytes.TrimSuffix(b, []byte(`"`))
+		b = bytes.ReplaceAll(b, []byte(`"`), []byte(`'`))
+		data := fmt.Sprintf(`{
+			"data": {
+				"type": "vars",
+				"attributes": {
+					"key":         "%s",
+					"value":       "%s",
+					"description": "%s",
+					"category":    "%s",
+					"hcl":         %s,
+					"sensitive":   %s
+				}
+			}
+		}
+		`, v.Name, string(b), v.Description, "terraform", "false", "false")
+		if payload == nil {
+			_, err := fmt.Fprintf(w, "%s", data)
+			payload = errors.Wrap(err, "tfvar: unexpected writing payload")
+		}
+	}
+
+	return payload
+}
+func WriteAsTFE_Resource(w io.Writer, vars []Variable) error {
+	f := hclwrite.NewEmptyFile()
+	rootBody := f.Body()
+
+	for _, v := range vars {
+		rootBody.AppendNewline()
+		resourceBlock := rootBody.AppendNewBlock("resource", []string{"tfe_variable", v.Name})
+		resourceBody := resourceBlock.Body()
```

Comment:
I think we are missing `key` argument here?
According to <https://registry.terraform.io/providers/hashicorp/tfe/latest/docs/resources/variable>, `key` is required

Reply:
yes, code blind :)